src/components/index.tsx

```diff
@@ -0,0 +1,146 @@
+import React, { useState, useImperativeHandle, forwardRef } from 'react';
+import { Editor, RichUtils, convertToRaw, convertFromRaw, EditorState } from 'draft-js';
+import { draftToMarkdown, markdownToDraft } from 'markdown-draft-js';
+import { IButtonStyles, IconButton, Stack } from '@fluentui/react';
+import { RichTextEditorProps } from '../interfaces';
+import { useEffectWithDebounce } from 'uno-react';
+
+const buttonStyles: IButtonStyles = {
+    icon: { color: 'grey' },
+    root: {
+        selectors: {
+            ':hover .ms-Button-icon': {
+                color: 'grey',
+            },
+            ':active .ms-Button-icon': {
+                color: 'grey',
+            },
+        },
+    },
+};
+const toMarkdownOptions = {
+    styleItems: {
+        UNDERLINE: {
+            open: () => '++',
+            close: () => '++',
+        },
+    },
+    preserveNewlines: true,
+};
+const fromMarkdownOptions = {
+    blockStyles: {
+        ins_open: ['UNDERLINE'],
+    },
+    remarkableOptions: {
+        enable: {
+            inline: ['ins'],
+        },
+    },
+    preserveNewlines: true,
+};
+const _convertFromMark = (markdown: string) => {
+    const raw = markdownToDraft(markdown, fromMarkdownOptions);
+    const content = convertFromRaw(raw);
+    return EditorState.createWithContent(content);
+};
+const _convertToMark = (editor: EditorState) => {
+    const content = editor.getCurrentContent();
+    const raw = convertToRaw(content);
+    const mark = draftToMarkdown(raw, toMarkdownOptions);
+    return mark;
+};
+
+export type RichTextEditorRef = {
+    updateUI: (newMarkdown: string) => void;
+};
+
+export const UnoReactDraftjs = forwardRef<RichTextEditorRef, RichTextEditorProps>(
+    ({ markdown, setMark, placeholder }: RichTextEditorProps, ref) => {
+        const [editorState, setEditor] = useState(_convertFromMark(markdown));
+        const [hidden, setHidden] = React.useState(true);
+        const [selected, setSelected] = React.useState({
+            bold: false,
+            italic: false,
+            underline: false,
+        });
+        const applyInlineStyle = (event: any, style: string) => {
+            event.preventDefault();
+            handleChange(RichUtils.toggleInlineStyle(editorState, style));
+            setSelected({
+                ...selected,
+                [style.toLowerCase()]: !selected[style.toLowerCase()],
+            });
+        };
+
+        const applyBlockType = (event: any, style: string) => {
+            event.preventDefault();
+            handleChange(RichUtils.toggleBlockType(editorState, style));
+        };
+
+        const handleChange = (newState: EditorState) => {
+            setEditor(newState);
+        };
```

Comment:
do we need this function? or we can call setEditor instead and delete handleChange?